FrontPage/ServiceProvider.swift

```diff
@@ -0,0 +1,126 @@
+        let keys = EmergenceKeys()
+        let auth = ArtsyAuthentication(clientID: keys.artsyAPIClientKey(), clientSecret: keys.artsyAPIClientSecret())
+        auth.getWeekLongXAppTrialToken(completion)
+    }
+
+    var topShelfItems: [TVContentItem] {
+        var items:[TVContentItem] = []
+        let semaphore = dispatch_semaphore_create(0)
+
+        auth { token, error in
+            // No auth, no go
+            if error != nil {
+                dispatch_semaphore_signal(semaphore)
+                return print("Error: \(error)")
+            }
+
+            // Setup the networking stack
+            let networkOp = ArtsyNetworkOperator()
+            let url = NSURL(string: "https://api.artsy.net/api/v1/set/530ebe92139b21efd6000071/items")!
+
+            // Make sure we apply our new xapp token
+            let request = NSMutableURLRequest(URL: url)
+            request.setValue(token.token, forHTTPHeaderField: "X-Xapp-Token")
+
+            let _ = networkOp.JSONTaskWithRequest(request, success: { request, response, json in
+
+                // When this block wraps up we can say we're done
+                defer { dispatch_semaphore_signal(semaphore) }
+
+                // This is a collection of shows raw JSON
+                guard let showDicts = json as? [[String:AnyObject]] else {
+                    return print("Could not convert the JSON to an array")
+                }
+
+                self.featuredShowsWrapper.topShelfItems = showDicts.map(self.showDictToWrapper)
+                items = [self.featuredShowsWrapper]
+
+            }, failure: { (request, response, error, json) in
+
+                // Less OK, but can't do much atm
+                dispatch_semaphore_signal(semaphore)
+            })
+        }
+
+        // wait for the above semaphore to complete before we go on
+        // if it locks forever, it's safe to assume apple's watchdog
+        // will kill the app
+        dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER)
+        return items
+    }
+
+    /// NSDict of a Show -> TVContentItem
+    func showDictToWrapper(show: [String:AnyObject]) -> TVContentItem {
+        let identifier = TVContentIdentifier(identifier: show["name"] as! String, container: self.featuredShowsWrapper.contentIdentifier)!
+        let contentItem = TVContentItem(contentIdentifier: identifier)!
+
+        // looks like: [___]
+        // they only support poster: [] and square [_]
+        // this seemed like the most reasonable fit
+
+        contentItem.imageShape = .HDTV
+
+        // So many `!` - but hey, swift hates JSON.
+        let showTitle = show["name"] as! String
+        let showID = show["id"] as! String
+        let partnerName = show["partner"]!["name"] as! String
+
+        contentItem.title = "\(showTitle) - \(partnerName)"
+        contentItem.imageURL = thumbnailForShowDict(show)
+        contentItem.displayURL = NSURL(string:"https://artsy.net/shows/\(showID)")
```

Comment:
roger that, though no web view here, it will get read in `func application(app: UIApplication, openURL url: NSURL, options: [String : AnyObject]) -> Bool ` and I'll ignore it. Better to have in though.